src/incremental/serialize.ml

```diff
@@ -38,22 +40,35 @@ let type_to_file_name = function
   | CilFile -> cil_file_name
   | VersionData -> version_map_filename
 
+let solver_data : Obj.t option ref = ref None
+
 (** Loads data for incremental runs from the appropriate file *)
 let load_data (data_type: incremental_data_kind) =
-  let p = Filename.concat (gob_results_dir ()) (type_to_file_name data_type) in
-  unmarshal p
+  if server () then
+    match data_type with
+    | SolverData -> !solver_data |> Option.get |> Obj.obj
+    | _ -> raise (Invalid_argument "Can only load solver data")
+  else
+    let p = Filename.concat (gob_results_dir ()) (type_to_file_name data_type) in
+    unmarshal p
 
 (** Stores data for future incremental runs at the appropriate file, given the data and what kind of data it is. *)
-let store_data (data: 'a) (data_type: incremental_data_kind) =
-  ignore @@ Goblintutil.create_dir (gob_directory ());
-  let d = gob_results_tmp_dir () in
-  ignore @@ Goblintutil.create_dir d;
-  let p = Filename.concat (d) (type_to_file_name data_type) in
-  marshal data p
+let store_data (data : 'a) (data_type : incremental_data_kind) =
+  if server () then
+    match data_type with
+    | SolverData -> solver_data := Some (Obj.repr data)
+    | _ -> ()
+  else (
```

Comment:
> Is init (i.e. Spec.init) called for the analyses before an incremental run in server mode? If init is called again, but without passing the data from the previous run, the data from the previous run might be overwritten and lost.

Yeah, ATM, `Spec.init None` is called in [control.ml](https://github.com/goblint/analyzer/blob/0aa80fd0ef3e6c7c9a198633809ce7100bffb498/src/framework/control.ml#L305) as usual since I don't set `load_run`. I'll fix it.

Reply:
Yes, #498 is not on master yet, I think you might just want to merge it into your branch.

Reply:
I think #498 could easily be merged into master or this PR. I think I didn't notice at the time that it was against interactive not master.

Reply:
I made a PR where I cherry-picked the relevant part of #498: https://github.com/goblint/analyzer/pull/537

Reply:
@keremc It's now merged into `master`, so you might want to merge `master` into this branch again.

Reply:
OK, I merged `master`.